Documentation/cilium_operator.rst

```diff
@@ -0,0 +1,212 @@
+
+For :ref:`concepts_ipam_crd` IPAM allocation mode, it is the job of custom
+operator, such as: "cilium-operator-aws", "cilium-operator-azure" to populate
+the required information about CIDRs in the ``CiliumNode`` resource.
+
+For more information on IPAM visit :ref:`address_management`.
+
+KVStore operations
+~~~~~~~~~~~~~~~~~~
+
+These operations are performed only when KVStore is enabled for the
+operator. In addition KVStore operations are only required when
+``cilium-operator`` is running with any of the below options:
+
+-  ``--synchronize-k8s-services``
+-  ``--synchronize-k8s-nodes``
+-  ``--identity-allocation-mode=kvstore``
+
+K8s Services synchronization
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Cilium operator performs the job of synchronizing Kubernetes services to
+external KVstore configured for the operator if running with
+``--synchronize-k8s-services`` flag.
+
+The operator performs this operation only for shared services(services
+that have ``io.cilium/shared-service`` annotation set to true). This is
+meaningful when running Cilium to setup a clustermesh.
+
+K8s Nodes synchronization
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Similar to K8s services, Cilium Operator also synchronize Kubernetes nodes
+information to the shared KVstore.
+
+When a ``Node`` object is deleted it is not possible to reliably cleanup
+the corresponding ``CiliumNode`` object from the agent itself. The operator
+holds the responsibility to garbage collect orphaned ``CiliumNodes``.
+
+CNP/CCNP node status GC
+^^^^^^^^^^^^^^^^^^^^^^^
+
+Similar to the effect of ``Node`` object deletion on ``CiliumNode``,
+Cilium Operator cannot remove the status corresponding to a node in a
+CNP/CCNP object. This operation of node status garbage collection from
+CNP/CCNP objects is also performed from the operator instead of the
+``cilium-agent``.
+
+This behavior can be disabled passing ``--set enableCnpStatusUpdates=false``
+to ``helm install`` when installing or updating Cilium:
+
+.. parsed-literal::
+
+    helm install cilium |CHART_RELEASE| \\
+      --namespace kube-system \\
+      --set enableCnpStatusUpdates=false
+
+Heartbeat update
+^^^^^^^^^^^^^^^^
+
+The operator periodically updates the Cilium's heartbeat path key
+with the current time. The default key for this heartbeat is
+``cilium/.heartbeat`` in the KVStore. It is used by agents to validate
+that kvstore updates can be received.
+
+Policy status update
+^^^^^^^^^^^^^^^^^^^^
+
+Cilium operator performs the operation of CNP/CCNP node status updates
+when ``k8s-events-handover`` is enabled. This is done to optimize
+Kubernetes events handling in case of large clusters. For the node
+status updates to be handled by ``cilium-operator``, all the K8s events
+are mirrored to the KVstore, which are then used to perform operations
+via the operator. This operation is performed for both
+CiliumNetworkPolicy and CiliumClusterwideNetworkPolicy objects.
+
+For each CNP/CCNP object in the cluster, we start a status handler. This
```

Comment:
Who is “we” in this sentence?

Reply:
I have updated such instances to specify `Cilium Operator`.